extensions/soundboard/src/soundform.tsx

```diff
@@ -2,19 +2,26 @@ import { Form, ActionPanel, Action, Icon, useNavigation } from "@raycast/api";
 import { Item } from "./types";
 import { useForm, FormValidation } from "@raycast/utils";
 import { nanoid } from "nanoid";
+import { useEffect, useState } from "react";
 
 interface SignUpFormValues {
   id: string;
   title: string;
   path: string[];
   favourite: string;
+  last_favourite: string;
 }
 
-export function SoundForm(props: { item?: Item; onEdit: (item: Item) => void }) {
+export function SoundForm(props: { item?: Item; items?: Item[]; onEdit: (item: Item) => void }) {
   const { pop } = useNavigation();
-  const { handleSubmit, itemProps } = useForm<SignUpFormValues>({
+  const [favoriteInUse, setFavoriteInUse] = useState<string>("");
+  const { handleSubmit, itemProps, values } = useForm<SignUpFormValues>({
     async onSubmit(values) {
-      props.onEdit(props?.item ? { ...values, id: props.item.id } : { ...values, id: nanoid() });
+      props.onEdit(
+        props?.item
+          ? { ...values, id: props.item.id, last_favourite: props.item.favourite }
+          : { ...values, id: nanoid() }
```

Comment:
If you wanted one less dependency, you could try using React's [useId](https://beta.reactjs.org/apis/react/useId) Hook instead of `nanoid`